### PR TITLE
Frontend Poll- and CreatePoll-page

### PR DESCRIPTION
**Pull request by QuayG**

### CODE REVIEW

File: frontend/src/pages/Menu.js [hunk 1]
```diff
@@ -18,7 +17,7 @@ export default function Menu() {
         <Page>
             <Header title="Menu"/>
             <Main>
-                <Avatar src={user.avatarUrl} alt=""/>
+                <h1>Hier könnte mal was stehen.</h1>
```
Comment (tobi-spec) [inline frontend/src/pages/Menu.js:20]: zum Beispiel IHRE Werbung!

**Review by tobi-spec — APPROVED** [review]

File: frontend/src/components/AnswerList.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import styled from "styled-components/macro";
+
+export default function AnswerList({possibleAnswers}) {
+    return (
+        <Wrapper>
+            {possibleAnswers.map(answer => (
+                <li key={possibleAnswers.indexOf(answer)}>{answer}</li>
+            ))}
```
Comment (jamarob) [inline frontend/src/components/AnswerList.js]: This is the same as:
```suggestion
            {possibleAnswers.map((answer,index) => (
                <li key={index}>{answer}</li>
            ))}
```
which is not so great, see here for reference: https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

Maybe you can find something else that is unique and  can be used as a key?

File: frontend/src/components/Avatar.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import styled from "styled-components/macro";
 
 export default styled.img`
-  width: 80%;
+  width: 50%;
   flex-grow: 1;
```
Comment (jamarob) [inline frontend/src/components/Avatar.js]: Why do you set the width but then allow it to grow?

File: frontend/src/components/Button.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ export default styled.button`
   color: var(--neutral-light);
   font-size: 1em;
   border-radius: var(--size-s);
+  margin: 10px;
```
Comment (jamarob) [inline frontend/src/components/Button.js:11]: Its better to give the parent element padding instead of a margin on the child. This makes your component more reusable. I doubt you want that margin around every button in your app?

File: frontend/src/components/Header.js [hunk 5]
```diff
@@ -10,10 +10,10 @@ export default function Header({title}){
     return(
         <Wrapper>
             {user && <Avatar src={user.avatarUrl} alt=""/>}
-            {!user && <div></div>}
-            <h1>{title}</h1>
+            {!user && <div/>}
+            <h3>{title}</h3>
             {user && <Button onClick={logout}>Logout</Button>}
-            {!user && <div></div>}
+            {!user && <div/>}
```
Comment (jamarob) [inline frontend/src/components/Header.js:16]: I don't get why you are conditionally closing the divs, looks a bit weird.
When I don't have a user the markup will look like this
```html
<header>
  </div>
  <h3>some title</h3>
  </div>
</header>
```
This is broken, please check again.

File: frontend/src/components/Header.js [hunk 6]
```diff
@@ -22,12 +22,14 @@ const Wrapper = styled.header`
   width: 100%;
   padding: var(--size-m);
 
-  text-align: center;
+  justify-items: center;
+  align-items: center;
```
Comment (jamarob) [inline frontend/src/components/Header.js:26]: Try a little bit of ordering for css, first display then the display related rules

File: frontend/src/components/Header.js [hunk 7]
```diff
@@ -22,12 +22,14 @@ const Wrapper = styled.header`
...
   display: grid;
   grid-template-columns: 30% 40% 30%;
   background: var(--background-dark);
   
   Button{
```
Comment (jamarob) [inline frontend/src/components/Header.js:30]:
```suggestion
  button{
```

File: frontend/src/pages/CreatePoll.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    const [possibleAnswers, setPossibleAnswers] = useState([])
+    const [loading, setLoading] = useState(false)
+    const [error, setError] = useState()
+
+    const handleSubmit = event => {
+        const poll = {
+            title: pollTitle,
+            user: user,
+            possibleAnswers: possibleAnswers
+        }
```
Comment (jamarob) [inline frontend/src/pages/CreatePoll.js:29]: Think about renaming `pollTitle` to `title` and then go
```javascript
const poll = {title, user, possibleAnswers}
```

File: frontend/src/pages/CreatePoll.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+        event.preventDefault()
+        setError()
+        setLoading(true)
+        createPoll(poll, token).catch(error => {
+            setError(error)
+        })
+        setLoading(false)
```
Comment (jamarob) [inline frontend/src/pages/CreatePoll.js:36]: You want to change this after the promise form `createPoll` has settled. Do it in a `then`, `catch` or `finally`.

File: frontend/src/pages/CreatePoll.js [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    const addPossibleAnswer = () => {
+        setPossibleAnswers([...possibleAnswers, answerToAdd])
+        setAnswerToAdd('')
+    }
+
+    if (!user) {
+        return <Redirect to="/"/>
+    }
```
Comment (jamarob) [inline frontend/src/pages/CreatePoll.js:56]: Works for now, but think about protected routes, simplifies your life a whole lot

File: frontend/src/pages/CreatePoll.js [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+  grid-template-rows: 10% 40% 10% 10% 10%;
+  place-items: center;
+  grid-gap: var(--size-xl);
+  padding: var(--size-xl);
+  margin: var(--size-xl);
+  height: 100%;
+  width: 100%;
+  overflow-y: scroll;
+  
+  Button{
```
Comment (jamarob) [inline frontend/src/pages/CreatePoll.js:100]:
```suggestion
  button{
```

When `Button` is a styled component, you can restyle it like so
```javascript
${Button} {
```

File: frontend/src/pages/Polls.js [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+    const [error, setError] = useState()
+
+    useEffect(() => {
+        if (token) {
+            setError()
+            setLoading(true)
+            findAll(token).then(setPolls).catch(error => {
+                setError(error)
+            })
+            setLoading(false)
```
Comment (jamarob) [inline frontend/src/pages/Polls.js]: same as before, do it when the promise has settled

File: frontend/src/pages/Polls.js [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+        return <Redirect to="/"/>
+    }
+
+    return (
+        <Page>
+            <Header title="Polls"/>
+            {loading && <Loading/>}
+            {!loading &&
+            <PollsList>
+                {polls && polls.map(poll => (
```
Comment (jamarob) [inline frontend/src/pages/Polls.js]: `polls` will always be an array, so you don't have to do the check before the map.

File: frontend/src/pages/Polls.js [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+                    <li key={poll.title}>
+                        <h3>{poll.title}</h3>
+                        <AnswerList>
+                            {poll.possibleAnswers && poll.possibleAnswers.map(answer => (
+                                <li key={answer}>{answer}</li>
+                            ))}
+                        </AnswerList>
+                        <p>No. of participants: {poll.givenAnswers.length}</p>
+                        <p>Created by: {poll.user.userName}</p>
+                    </li>
```
Comment (jamarob) [inline frontend/src/pages/Polls.js]: I feel like there is a component for a single poll hidden somewhere :)

File: frontend/src/pages/Polls.js [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+      margin: 0;
+      padding: 0;
+
+      li {
+        justify-content: center;
+        width: auto;
+        margin: 15px;
+        padding: 15px;
+      }
+    }
```
Comment (jamarob) [inline frontend/src/pages/Polls.js]: Hm.. better put those styles in the `AnswerList`, or restyle it here. This nesting is hard to follow as it is not immediately clear where the inner ul and li come from

File: frontend/src/pages/Register.js [hunk 16]
```diff
@@ -38,6 +42,7 @@ export default function Register(){
         }
         createUser(user).then(()=>setCreatedUser(user)).catch(error=>{
             setError(error)
+            setLoading(false)
```
Comment (jamarob) [inline frontend/src/pages/Register.js:45]: 👍

**Review by jamarob — COMMENTED** [review]
good job! 👍 easy to read and review! I left some comments, I think your Header needs an inspection ;)

File: frontend/src/components/Header.js (hunk 5 above)
Comment (QuayG) [reply frontend/src/components/Header.js:16]: It's a conditional rendering of empty divs. I did that to avoid the title move to the left grid box, in case that there is no user and therefor no avatar to be rendered in the left grid box
